### PR TITLE
feat(engine): add transoforms to name policies

### PR DESCRIPTION
**Pull request by W95Psp**

### CODE REVIEW

**Review by franziskuskiefer — APPROVED** [review]
lgtm. Let's wait for review from @jschneider-bensch before merging.

**Comment by W95Psp** [comment]
Uh, I think I included too much stuff in my commit, I need to rebase it

File: engine/lib/ast_utils.ml [hunk 1]
```diff
@@ -226,6 +226,91 @@ module Make (F : Features.T) = struct
...
+            | _ -> { e; span; typ }
+        end
+      in
+      obj#visit_expr ()
+
+    let insert_prop_coercions : (module ERROR) -> expr -> expr =
+      let lift_fun _f = None in
+      let lift_arg _arg ~goal:_ = None in
+      fun err -> insert_coercions err lift_fun lift_arg
+
```
Comment (jschneider-bensch) [inline engine/lib/ast_utils.ml:313]: This is probably unrelated to the name policy, right?

File: engine/lib/concrete_ident/names.rs [hunk 2]
```diff
@@ -112,6 +112,7 @@ fn unsize() {}
 mod hax {
     fn failure() {}
     struct Failure;
+    struct Prop;
```
Comment (jschneider-bensch) [inline engine/lib/concrete_ident/names.rs:115]: Also unrelated?

**Review by jschneider-bensch — COMMENTED** [review]
Thanks for this! I tested it out with my current branch of the PV backend :+1:  and it works a charm for simplifying the naming of struct and enum constructors and fields.

I noticed that `fn`s with no arguments now received a dummy argument `_ : ()`. Could that be related to the other changes in the PR?

File: engine/lib/ast_utils.ml (hunk 1 above)
Comment (W95Psp) [reply engine/lib/ast_utils.ml:313]: Yes, exactly! :sweat_smile:

**Comment by W95Psp** [comment]
The `()` thing is implemented by #388, so that now every Rust function is always translated as a function of non-zero arity in Hax's AST

**Review by jschneider-bensch — APPROVED** [review]
Thanks, this looks good to me!

**Pull request merged**